lib/Handler.php

```diff
@@ -310,4 +310,54 @@ protected function notificationFromRow(array $row) {
 
 		return $notification;
 	}
+
+	/**
+	 * Remove the base url from absolute links in the database.
+	 * This affects the columns 'link' and 'action'.
+	 * e.g: http://owncloud.com/test -> /test
+	 *
+	 * @return int number of updated notifications
+	 */
+	public function removeBaseUrlFromAbsoluteLinks() {
+		$sql = $this->connection->getQueryBuilder();
+		$sql->select(['notification_id', 'link', 'actions'])
+			->from('notifications')
+			->where($sql->expr()->like('link', $sql->createPositionalParameter('http%')))
+			->orWhere($sql->expr()->like('actions', $sql->createPositionalParameter('%"link":"http%')));
+
+		$statement = $sql->execute();
+		$counter = 0;
+
+		while ($row = $statement->fetch()) {
+			$sql = $this->connection->getQueryBuilder();
+			$sql->update('notifications')
+				->where($sql->expr()->eq('notification_id', $sql->createNamedParameter($row['notification_id'])));
+
+			$linkUrlComponents = \parse_url($row['link']);
+			if (isset($linkUrlComponents['scheme'])) {
+				$newLink = \parse_url($row['link'], PHP_URL_PATH);
```

Comment:
`$newLink = $linkUrlComponents['path']` seems a better choice to avoid parsing the same url again